### PR TITLE
Add openvino export and supported models sections

### PR DESCRIPTION
**Pull request by echarlaix**

### CODE REVIEW

**Review by helena-intel — APPROVED** [review]
Thanks @echarlaix this looks great!

File: tests/openvino/test_modeling.py [hunk 1]
```diff
@@ -1320,17 +1324,17 @@ def test_compare_with_and_without_past_key_values(self):
...
-        # "data2vec_audio",
-        # "hubert",
-        # "sew",
-        # "sew_d",
-        # "wav2vec2-conformer",
+        "audio_spectrogram_transformer",
+        "data2vec_audio",
+        "hubert",
+        "sew",
+        "sew_d",
```
Comment (IlyasMoutawwakil) [inline tests/openvino/test_modeling.py:1331]: I see a couple of new errors in the CI related to these models. Is there a reason why they're uncommented in this PR ?

**Review by AlexKoff88 — APPROVED** [review]

**Review by IlyasMoutawwakil — APPROVED** [review]

File: docs/source/openvino/architectures.mdx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+🤗 Optimum handles the export of models to OpenVINO in the `exporters.openvino` module. It provides classes, functions, and a command line interface to perform the export easily.
+Here is the list of the supported architectures :
+
+## [Transformers](https://huggingface.co/docs/transformers/index):
+
+- Albert
+- Albert
+- Aquila
+- Aquila
```
Comment (eaidova) [inline docs/source/openvino/architectures.mdx:20]: it is expected that some models listed twice?
```suggestion
- Albert
- Aquila
```

Comment (echarlaix) [reply docs/source/openvino/architectures.mdx]: good catch !

File: optimum/commands/export/openvino.py [hunk 3]
```diff
@@ -72,31 +69,40 @@ def parse_args_openvino(parser: "ArgumentParser"):
...
-    optional_group.add_argument("--fp16", action="store_true", help="Compress weights to fp16")
-    optional_group.add_argument("--int8", action="store_true", help="Compress weights to int8")
     optional_group.add_argument(
-        "--weight-format",
+        "--library",
         type=str,
-        choices=["fp32", "fp16", "int8", "int4", "int4_sym_g128", "int4_asym_g128", "int4_sym_g64", "int4_asym_g64"],
+        choices=["transformers", "diffusers", "timm", "sentence_transformers"],
+        default=None,
+        help="The library used to laod the model before export. If not provided, will attempt to infer the local checkpoint's library",
```
[the remark this answers is not in the document]
Comment (echarlaix) [reply optimum/commands/export/openvino.py:83]: good catch !

File: tests/openvino/test_modeling.py (hunk 1 above)
Comment (echarlaix) [reply tests/openvino/test_modeling.py:1331]: added them to know what we can have in the supported architectures [section]( https://moon-ci-docs.huggingface.co/docs/optimum-intel/pr_765/en/openvino/architectures)

File: docs/source/openvino/inference.mdx [hunk 4]
```diff
@@ -7,81 +7,13 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
 ## Transformers models
 
-You can now easily perform inference with OpenVINO Runtime on a variety of Intel processors
-([see](https://docs.openvino.ai/2024/about-openvino/compatibility-and-support/supported-devices.html) the full list of supported devices). 
-For that, just replace the `AutoModelForXxx` class with the corresponding `OVModelForXxx` class.
-
-As shown in the table below, each task is associated with a class enabling to automatically load your model.
-
-| Task                                 | Auto Class                           |
```
[the remark this answers is not in the document]
Comment (echarlaix) [reply docs/source/openvino/inference.mdx:22]: added in https://github.com/huggingface/optimum-intel/pull/765/commits/6896d1eaa5289145072bade81568b183cbc06500

File: optimum/commands/export/openvino.py [hunk 5]
```diff
@@ -72,31 +69,40 @@ def parse_args_openvino(parser: "ArgumentParser"):
...
         default=None,
         help=(
-            "The weight format of the exporting model, e.g. f32 stands for float32 weights, f16 - for float16 weights, i8 - INT8 weights, int4_* - for INT4 compressed weights."
+            "This is needed by some models, for some tasks. If not provided, will attempt to use the tokenizer to guess it."
         ),
     )
     optional_group.add_argument(
         "--ratio",
         type=float,
         default=None,
```
[the remark this answers is not in the document]
Comment (echarlaix) [reply optimum/commands/export/openvino.py:102]: added in https://github.com/huggingface/optimum-intel/pull/765/commits/d60b8ed947867866226d9d879cb56f575e236276

**Pull request merged**